feat: Design System Publisher documentation (#394)
vaadin/docs: articles/tools/dspublisher/customization.asciidoc

order: 30
---

= Site Customization

The documentation website theme can be customized to fit the brand and design guidelines of your organization.
A <<custom-theme#,custom design system theme>> can also be applied, which affects the theme of rendered UI examples within the documentation pages.

== Website Theme

The website's theme can be defined in the [filename]#global.css# stylesheet in the [filename]#dspublisher/docs-theme# folder.
Most styles can be customized using custom properties.
These are the most commonly used style properties:

[cols=2,frame=none,grid=rows]
|===
|`--docs-font-family`
|Font family

|`--docs-font-family-heading`
|Font family for headings

|`--docs-font-size-h1...h6`
|Font size for H1...H6 level headings

|`--docs-body-text-color`
|Color of body text

|`--docs-heading-text-color`
|Color of headings

|`--docs-link-color`
|Color of unvisited links

|`--docs-visited-link-color`
|Color of visited links

|`--docs-header-background-color`
|Header background color

|`--docs-tab-selected-color`
|Selected tab color (for code sample tabs, etc)

|`--docs-layout-max-width`
|Max width of the website's content

|`--docs-article-sidebar-width`

> |Width of the page sidebar / TOC

Author: Or just "table of contents" in full

---
title: Site Customization
order: 50
---

= Site Customization

The documentation website styles can be customized to fit the brand and design guidelines of your organization.

> A <<custom-theme#,custom design system theme>> can also be applied, which affects the theme of rendered UI examples within the documentation pages.

Author: ```suggestion
A <<custom-theme#,custom application theme>> can also be applied, which affects the theme of rendered UI examples within the documentation pages.
```

.`dspublisher/docs-theme/global.css`
[source,css]
----
html {
  --docs-header-background-color: green;
  --docs-font-family: 'Times New Roman';
}
----

Changes to the website's styles are automatically applied after a couple of seconds.

=== Header

The “full name” given to the initialization script sets the website's title.
See <<settings>> for changing it after initialization.

The title text size is determined by the `--docs-font-size-h3` property, but its styling can be fully customized as follows:

.`dspublisher/docs-theme/global.css`
[source,css]
----
header h3 {
  --docs-heading-text-color: red;
  --docs-font-family-heading: Helvetica;
}
----

A logo image can be inserted into website's header by copying the image into the [filename]#docs-theme# folder and replacing the `background-image` value in the header style block.

Depending on the size and proportions of the image, you may also need to set the `background-size` and `background-repeat` properties as follows:

.`dspublisher/docs-theme/global.css`
[source,css]
----
header h3 a::before {
  background-image: url("logo.png");
  background-size: contain;
  background-repeat: no-repeat;
}
----

> .Design system theme

Author: ```suggestion
.Application theme
```

[NOTE]
This section is about styling the documentation website itself.

> See <<custom-theme#,Applying a Custom Design System Theme>> for instructions how to apply a theme which is used in <<ui-examples#,UI examples>>.

Author: ```suggestion
See <<custom-theme#,Applying a Custom Application Theme>> for instructions how to apply a theme which is used in <<ui-examples#,UI examples>>.
```

---
title: Site Customization
order: 30
---

= Site Customization

The documentation website theme can be customized to fit the brand and design guidelines of your organization.
A <<custom-theme#,custom design system theme>> can also be applied, which affects the theme of rendered UI examples within the documentation pages.

== Website Theme

> The website's theme can be defined in the [filename]#global.css# stylesheet in the [filename]#dspublisher/docs-theme# folder.

Author: ```suggestion
The website's theme can be defined in the [filename]#global.css# style sheet in the [filename]#dspublisher/docs-theme# folder.
```

Most styles can be customized using custom properties.

The following are the most commonly used style properties:

[cols=2,frame=none,grid=rows]
|===
|`--docs-font-family`
|Font family

|`--docs-font-family-heading`
|Font family for headings

|`--docs-font-size-h1...h6`
|Font size for H1...H6 level headings

|`--docs-body-text-color`
|Color of body text

|`--docs-heading-text-color`
|Color of headings

|`--docs-link-color`
|Color of unvisited links

|`--docs-visited-link-color`
|Color of visited links

|`--docs-header-background-color`
|Header background color

|`--docs-tab-selected-color`
|Selected tab color (for code sample tabs, etc)

|`--docs-layout-max-width`
|Max width of the website's content

|`--docs-article-sidebar-width`
|Width of the page sidebar / table of contents

|`--docs-article-max-width`

> |Max width of the page's body (excluding sidebar/TOC)

Author: ```suggestion
|Max width of the page's body (excluding sidebar/table of contents)
```

=== Header

The “full name” given to the initialization script sets the website's title.
See <<settings>> for changing it after initialization.

The title text size is determined by the `--docs-font-size-h3` property, but its styling can be fully customized as follows:

.`dspublisher/docs-theme/global.css`
[source,css]
----
header h3 {
  --docs-heading-text-color: red;
  --docs-font-family-heading: Helvetica;
}
----

A logo image can be inserted into website's header by copying the image into the [filename]#docs-theme# folder and replacing the `background-image` value in the header style block.

Depending on the size and proportions of the image, you may also need to set the `background-size` and `background-repeat` properties as follows:

.`dspublisher/docs-theme/global.css`
[source,css]
----
header h3 a::before {
  background-image: url("logo.png");
  background-size: contain;
  background-repeat: no-repeat;
}
----

.Component theme
[NOTE]
This section is about theming the documentation website itself.
See <<custom-theme#,Applying a Custom Design System Theme>> for instructions how to apply a theme is used in <<ui-examples#,UI examples>>.

== Custom Top Header

> There is an empty top-level header element (`<dspublisher-header>`) on the documentation site, above the application header.
> You can add custom content inside the top header using the "[filename]#dspublisher/docs-theme/header.ts#" file.

Author: ```suggestion
You can use the empty top-level header element (`<dspublisher-header>`) on the documentation site, above the application header, to add custom content inside it using the "[filename]#dspublisher/docs-theme/header.ts#" file.
```
